EnvironmentSimulator/Modules/RoadManager/RoadManager.cpp

```diff
@@ -3777,6 +4520,146 @@ void RMObject::SetRepeat(Repeat* repeat)
+	//object.append_attribute("dynamic").set_value() TODO:
+	object.append_attribute("name").set_value(name_.c_str());
+	object.append_attribute("id").set_value(id_);
+	object.append_attribute("s").set_value(s_);
+	object.append_attribute("t").set_value(t_);
+	object.append_attribute("zOffset").set_value(z_offset_);
+	//object.append_attribute("validLength").set_value(); TODO:
+	switch (orientation_)
+	{
+	case RoadObject::Orientation::NEGATIVE :
+		object.append_attribute("orientation").set_value("-");
+		break;
+	case RoadObject::Orientation::POSITIVE :
+		object.append_attribute("orientation").set_value("+");
+		break;
+	case RoadObject::Orientation::NONE :
+		object.append_attribute("orientation").set_value("none");
+		break;
+	default:
+		break;
+	}
+	object.append_attribute("hdg").set_value(heading_);
+	object.append_attribute("pitch").set_value(pitch_);
+	object.append_attribute("roll").set_value(roll_);
+	if(length_)
+		object.append_attribute("length").set_value(length_);
+	if(width_)
+		object.append_attribute("width").set_value(width_);
+	if(height_)
+		object.append_attribute("height").set_value(height_);
+
+	if(GetRepeat() != nullptr)
+	{
+		GetRepeat()->Save(object);
+	}
+
+
+	for(auto outline : outlines_)
+	{
+		outline->Save(object);
+	}
+
+	for(auto validity : validity_)
+	{
+		validity.Save(object);
+	}
+}
+
+void ValidityRecord::Save(pugi::xml_node& object)
+{
+	auto validity = object.append_child("validity");
+	validity.append_attribute("fromLane").set_value(fromLane_);
+	validity.append_attribute("toLane").set_value(toLane_);
+}
+
+
+
+void Outline::Save(pugi::xml_node& object)
+{
+	auto outlines = object.child("outlines");
+	if(outlines.empty())
+	{
+		outlines = object.append_child("outlines");
+	}
+	auto outline = outlines.append_child("outline");
+
+	if(id_)
+		outline.append_attribute("id").set_value(id_);
+
+	switch (fillType_)
+	{
+	case Outline::FillType::FILL_TYPE_GRASS : 
+		outline.append_attribute("fillType").set_value("grass");
+		break;
+	case Outline::FillType::FILL_TYPE_CONCRETE : 
+		outline.append_attribute("fillType").set_value("concrete");
+		break;
+	case Outline::FillType::FILL_TYPE_COBBLE : 
+		outline.append_attribute("fillType").set_value("cobble");
+		break;
+	case Outline::FillType::FILL_TYPE_ASPHALT : 
+		outline.append_attribute("fillType").set_value("asphalt");
+		break;
+	case Outline::FillType::FILL_TYPE_PAVEMENT : 
+		outline.append_attribute("fillType").set_value("pavement");
+		break;
+	case Outline::FillType::FILL_TYPE_GRAVEL : 
+		outline.append_attribute("fillType").set_value("gravel");
+		break;
+	case Outline::FillType::FILL_TYPE_SOIL : 
+		outline.append_attribute("fillType").set_value("soil");
+		break;
+	default:
+		break;
+	}
+	
+	//outline.append_attribute("outer").set_value();
+	if(closed_)
+		outline.append_attribute("closed").set_value("true");
+	else if(!closed_)
+		outline.append_attribute("closed").set_value("false");
+
+	//outline.append_attribute("laneType").set_value();
```

Comment:
remove

Reply:
These kind of commented out stuff I left since there are no parser for these attributes. Will add TODO mark aswell.

Reply:
:+1: